### ISSUE TITLE
Positioning architecture research

### ISSUE THREAD

**Issue by kade-ddnkv** [edgecomllc/eupf#470]
![Image](https://user-images.githubusercontent.com/20152142/278684069-4f5ef266-7142-4ee5-9f20-c7dc539c7c26.png)

Acceptance criteria:

- [ ] Explore 5G 3GPP location architecture (spec link [will be here](https://www.ericsson.com/en/blog/2020/12/5g-positioning--what-you-need-to-know) and [here](https://www.mdpi.com/1424-8220/22/13/4757)) 
- [ ] Understand which components are responsible for location based services, what kind of data is needed
- [ ] Check whether UERANSIM/srsRAN/OAI have enough functions to emulate location tracking services

Useful links:

- https://forum.huawei.com/enterprise/en/5G-positioning-Introduction/thread/667270971930853377-667213872962088960

**Comment by kade-ddnkv** [comment]
**Scheme**  
NG-C interface (connects CU-CP to the AMF)  
\> NGAP (on top of SCTP using ASN1 messages)  
\>> NRPPa transport

![image](https://github.com/edgecomllc/eupf/assets/88973774/41e55f8e-72ee-45ef-b742-1f6721d161d0)
Fig. 1: Architecture. (Source: URL...)  

![image](https://github.com/edgecomllc/eupf/assets/88973774/bd8e23bb-4a30-45a8-a2c3-6dc45e7f25ab)
Fig. 2: NRPPa sequence diagram. (Source: URL...)  

$$~$$  


**What protocols/signals are supported in emulators:** 
|                  | NRPPa transport | NRPPa functionalities | PRS           |
| ---------------- | --------------- | --------------------- | ------------- |
| UERANSIM         | ? (code)        | ? (code)              | ? (code)      |
| srsRAN           | +               | ? (code)              | +             |
| OpenAirInterface | +               | OTDOA +, E-CID ?      | + (no muting) |

Вывод: без документации невкусно и запетая

$$~$$  

**Positioning methods classification:**  
* *Cell-Identity-Based* - relies on radio coverage and serving gNB location.
* *Angle-Based* - uses direction of a signal received by antenna (arrival or departure angle). Especially effective with massive MIMO.
* *Range-Based* - uses measurememts between UE and gNB such as RSS, ToA, TDoA. Needs at least 3 gNB to use triliteration.
* *Fingerprinting-Based* - records measurements from points in real world, then trains ML model. Effective in Non-Line-of-Sight environment. 

$$~$$  

**What data can be retieved from emulated gNB-s:** 
|                  | Angle of Arrival | Time (Difference) of Arrival | RSRP     | beam-RSRP | CSI      | RSS            |
| ---------------- | ---------------- | ---------------------------- | -------- | --------- | -------- | -------------- |
| UERANSIM         | ?                | ?                            | ?        | ?         | ?        | ?              |
| srsRAN           | ? (code)         | ? (code)                     | + (code) | - (code)  | + (code) | - (code, RSSI) |
| OpenAirInterface | +?               | +                            | +        | - (code)  | +        | - (code, RSSI) |

$$~$$  

**In conclusion, what positioning methods can we use?**  
|                  | Angle-Based                              | Range-Based | ML-Aided |
| ---------------- | ---------------------------------------- | ----------- | -------- |
| UERANSIM         | ?                                        | ?           | ?        |
| srsRAN           | probably -                               |     probably +        | +        |
| OpenAirInterface | probably - (NRPPa E-CID not implemented) | +           | +        |

$$~$$  

**Unresolved issues:**
* Can I define UE location in emulator?

**Comment by kade-ddnkv** [comment]
OpenAirInterface Status Presentation (25.05.2023): https://openairinterface.org/wp-content/uploads/2023/06/9a-Walkthrough-RAN-Demos.pdf
Positioning experiment using PRS from OpenAirInterface: https://link.springer.com/article/10.1007/s12243-023-00987-2
